framework/include/systems/NonlinearSystemBase.h

```diff
@@ -556,6 +556,21 @@ class NonlinearSystemBase : public SystemBase, public PerfGraphInterface
 
   Moose::MooseKSPNormType getMooseKSPNormType() { return _ksp_norm; }
 
+  /**
+   * We offer the option to check convergence against the residual norm \em prior to executing
+   * solution modifying objects. This method handles the logic as to whether we should perform such
+   * residual evaluation.
+   *
+   * \return A boolean indicating whether we should evaluate the pre-initial residual.
+   */
+  bool shouldEvaluatePreinitialResidual() const;
```

Comment:
```suggestion
  bool shouldEvaluateResidualBeforeSolutionModifyingObjects() const;
```
how about that for a mouthful! Again I think something like this is more clear as I would argue that if you do evaluate this residual, then it is chronologically the initial residual